add realtime speaker identification documentation (#113)
speechmatics/docs: docs/speech-to-text/features/speaker-identification.mdx

---
description: "Learn how Speechmatics identifies speakers in audio"
keywords:
  [
    speechmatics,
    features,
    speaker identification,
    diarization,
    transcription,
    speech recognition,
    automatic speech recognition,
    asr,
  ]
sidebar_position: 2
---

# Speaker identification

Speaker identification builds on diarization by recognizing specific speakers in audio. Rather than labeling voices generically (S1, S2, etc.), it lets you tag them consistently across recordings with the help of speaker identifiers, which are string-encoded speakers voice representations generated from short audio samples.

> It’s particularly useful when the same speakers appear across multiple recordings, such as in recurring meetings, news feeds, or customer support calls. By tagging known speakers with consistent labels, speaker identification makes transcripts more accurate, searchable, and easier to analyze over time.

Reviewer: We could potentially also mention that "Initialising the search space with trained speakers can also increase the accuracy of the system by providing established speaker information based on reliable and task appropriate data." (or similar?)

## Use cases

- **Meetings and interviews** — Tag recurring participants across sessions.

> - **Media and news** — Consistently label public figures.

Reviewer: Not sure if we want to expand, but to the question of "why" we'd label public figures, we could add _", which could be valuable in, for example, subtitling, media search and archiving."_

> - **Customer support** — Distinguish agents from customers by name.

Reviewer: Looks good, thanks.

## How it works

To use speaker identification you must enable diarization in one of the two supported modes —
   [speaker diarization](../features/diarization.mdx#speaker-diarization) or
   [channel & speaker diarization](../features/diarization.mdx#channel--speaker-diarization) and then follow the two steps below:

- **Enrollment** — For each speaker you want to recognize, generate identifiers from short audio clips (5–30s) where they ideally speak alone.

>   To improve robustness, you can enroll the same speaker with multiple clips recorded under different acoustic conditions.

Reviewer: ...different acoustic conditions, selected to represent the degree of variety and quality that could be expected in the target audio.

(or similar?)

- **Video conferences** – Automatically label participants across multiple meetings to maintain consistent speaker records and analytics.
- **Medical consultations** – Identify doctors and patients across sessions for accurate records and follow-up care.
- **Media production** – Consistently label recurring speakers or public figures across episodes or segments, which could be valuable in, for example, subtitling, media search and archiving.

## How it works

To use speaker identification you must enable diarization in one of the two supported modes —
   [speaker diarization](../features/diarization.mdx#speaker-diarization) or
   [channel & speaker diarization](../features/diarization.mdx#channel--speaker-diarization) and then follow the two steps below:

- **Enrollment** — For each speaker you want to recognize, generate identifiers from short audio clips (5–30s) where they ideally speak alone.
  To improve robustness, you can enroll the same speaker with multiple clips recorded under different acoustic conditions, selected to represent the degree of variety and quality that could be expected in the target audio.

:::warning

> Speaker identifiers are tied to the model used to generated them, and therefore are specific to the operating point in which they were created. Using identifiers across operating points is not supported and any such identifiers will be ignored. Whenever the model within an operating point is updated, existing identifiers must always be regenerated.

Reviewer 2: `Using identifiers across operating points is not supported and any such identifiers will be ignored.` -> Here I think we do ignore but also send a warning message, so we update indicating that?

---
description: "Learn how Speechmatics identifies speakers in audio"
keywords:
  [
    speechmatics,
    features,
    speaker identification,
    diarization,
    transcription,
    speech recognition,
    automatic speech recognition,
    asr,
  ]
sidebar_position: 2
---

# Speaker identification

> Speaker identification builds on diarization by recognizing specific speakers in audio. Rather than labeling voices generically (S1, S2, etc.), it lets you tag them consistently across recordings with the help of speaker identifiers, which are string-encoded speakers voice representations generated from short audio samples.

Reviewer 3: ```suggestion
Speaker identification lets you tag speakers consistently across recordings with the help of speaker identifiers. You can generate these string-encoded voice representations using short audio samples of the target speakers.
```

> It’s particularly useful when the same speakers appear across multiple recordings or streams, such as in recurring meetings, news feeds, or customer support calls. By tagging known speakers with consistent labels, speaker identification makes transcripts more accurate, searchable, and easier to analyze over time. In addition, initialising the search space with trained speakers can also increase the accuracy of the system by providing established speaker information based on reliable and task appropriate data.

Reviewer 3: ```suggestion
By tagging known speakers with consistent labels, speaker identification makes transcripts more accurate, searchable, and easier to analyze over time. Providing speaker identifiers can also increase that accuracy of our diarization system.
```

Author: applied, thanks

## Use cases

- **Voice agents/call centers** – Recognize and tag individual agents or speakers by name for personalized service, training, and compliance tracking.

> - **Video conferences** – Automatically label participants across multiple meetings to know who said what, maintain consistent speaker records and analytics.

Reviewer 3: ```suggestion
- **Video conferences** – Automatically label participants across multiple meetings to know who said what and maintain consistent speaker records and analytics.
```

- **Medical consultations** – Identify doctors or patients across sessions for accurate records, summaries and follow-up care.

> - **Media production** – Consistently label recurring speakers or public figures across episodes or segments, which could be valuable in, for example, subtitling, media search and archiving.

Reviewer 3: ```suggestion
- **Media production** – Consistently label recurring speakers or public figures across episodes or segments, which is valuable in subtitling, media search and archiving.
```

Author: applied, thanks

## How it works

To use speaker identification you must enable diarization in one of the two supported modes —
   [speaker diarization](../features/diarization.mdx#speaker-diarization) or
   [channel & speaker diarization](../features/diarization.mdx#channel--speaker-diarization) and then follow the two steps below:

- **Enrollment** — For each speaker you want to recognize, generate identifiers from short audio clips (5–30s) where they ideally speak alone.
  To improve robustness, you can enroll the same speaker with multiple clips recorded under different acoustic conditions, selected to represent the degree of variety and quality that could be expected in the target audio.

:::warning
Speaker identifiers are tied to the model used to generated them, and therefore are specific to the operating point in which they were created. Using identifiers across operating points is not supported and any such identifiers will be ignored. 
:::
:::warning
With some updates, existing speaker identifiers may need to be regenerated, this will require the speaker audio. A warning will be raised in the API. Jobs will complete, but will no longer have the speaker label until they are regenerated.

> :::

Reviewer 3: You may want to move these into a separate best practices/gotchas section to call out some issues people are likely to encounter with the system

Author: good idea, moved to the separate "Known limitations and caveats" section

## How it works

To use speaker identification you must enable diarization in one of the two supported modes —
   [speaker diarization](../features/diarization.mdx#speaker-diarization) or
   [channel & speaker diarization](../features/diarization.mdx#channel--speaker-diarization) and then follow the two steps below:

- **Enrollment** — For each speaker you want to recognize, generate identifiers from short audio clips (5–30s) where they ideally speak alone.
  To improve robustness, you can enroll the same speaker with multiple clips recorded under different acoustic conditions, selected to represent the degree of variety and quality that could be expected in the target audio.

- **Identification** — Use the enrolled identifiers in transcription jobs to label known speakers with meaningful names (for example, `Alice` or `John`).  The system matches voices to identifiers and tags the output with the desired labels.

:::info
Labels for identified speakers must not use reserved internal labels (UU, S1, S2, etc.) and must not include leading or trailing spaces.
:::

> ## Known limitations and caveats

Reviewer 3: Can we call this something different? Limitations sounds so negative haha

Author: used "Known caveats" for now
